src/jdk.jshell/share/classes/jdk/jshell/execution/LocalExecutionControl.java

```diff
@@ -57,8 +66,61 @@ public LocalExecutionControl(LoaderDelegate loaderDelegate) {
     public LocalExecutionControl() {
     }
 
+    @Override
+    public void load(ClassBytecodes[] cbcs)
+            throws ClassInstallException, NotImplementedException, EngineTerminationException {
+        super.load(Stream.of(cbcs)
+                .map(cbc -> new ClassBytecodes(cbc.name(), instrument(cbc.bytecodes())))
+                .toArray(ClassBytecodes[]::new));
+    }
+
+    private static byte[] instrument(byte[] classFile) {
+        var reader  = new ClassReader(classFile);
+        var writer = new ClassWriter(reader, 0);
+        reader.accept(new ClassVisitor(Opcodes.ASM9, writer) {
+            @Override
+            public MethodVisitor visitMethod(int access, String name, String descriptor, String signature, String[] exceptions) {
+                return new MethodVisitor(Opcodes.ASM9, super.visitMethod(access, name, descriptor, signature, exceptions)) {
+                    @Override
+                    public void visitJumpInsn(int opcode, Label label) {
+                        visitMethodInsn(Opcodes.INVOKESTATIC, "REPL/$Cancel$", "stopCheck", "()V", false);
+                        super.visitJumpInsn(opcode, label);
+                    }
+                };
+            }
+        }, 0);
+        return writer.toByteArray();
+    }
+
+    private static ClassBytecodes genCancelClass() {
+        var cancelWriter = new ClassWriter(ClassWriter.COMPUTE_FRAMES|ClassWriter.COMPUTE_MAXS);
+        cancelWriter.visit(Opcodes.V19, Opcodes.ACC_PUBLIC, "REPL/$Cancel$", null, "java/lang/Object", null);
+        cancelWriter.visitField(Opcodes.ACC_PUBLIC|Opcodes.ACC_STATIC|Opcodes.ACC_VOLATILE, "allStop", "Z", null, null);
+        var checkVisitor = cancelWriter.visitMethod(Opcodes.ACC_PUBLIC|Opcodes.ACC_STATIC, "stopCheck", "()V", null, null);
+            checkVisitor.visitCode();
+            checkVisitor.visitFieldInsn(Opcodes.GETSTATIC, "REPL/$Cancel$", "allStop", "Z");
+            var skip = new Label();
+            checkVisitor.visitJumpInsn(Opcodes.IFEQ, skip);
+            checkVisitor.visitTypeInsn(Opcodes.NEW, "java/lang/ThreadDeath");
+            checkVisitor.visitInsn(Opcodes.DUP);
+            checkVisitor.visitMethodInsn(Opcodes.INVOKESPECIAL, "java/lang/ThreadDeath", "<init>", "()V", false);
+            checkVisitor.visitInsn(Opcodes.ATHROW);
+            checkVisitor.visitLabel(skip);
+            checkVisitor.visitInsn(Opcodes.RETURN);
+            checkVisitor.visitMaxs(0, 0);
+            checkVisitor.visitEnd();
+        cancelWriter.visitEnd();
+        return new ClassBytecodes("REPL.$Cancel$", cancelWriter.toByteArray());
+    }
+
     @Override
     protected String invoke(Method doitMethod) throws Exception {
+        if (allStop == null) {
+            super.load(new ClassBytecodes[]{genCancelClass()});
```

Comment:
fomatting: usually, people use spaces like this
```
  new ClassBytecodes[] {  genCancelClass()  }
```

Reply:
Fixed, thanks